src/views/Bond/hooks/useBondTokens.ts

```diff
@@ -0,0 +1,88 @@
+import { OHMTokenProps } from "@olympusdao/component-library";
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+import { ethers } from "ethers";
+import { OHM_ADDRESSES } from "src/constants/addresses";
+import { DecimalBigNumber } from "src/helpers/DecimalBigNumber/DecimalBigNumber";
+import { useTestableNetworks } from "src/hooks/useTestableNetworks";
+import { BondFixedExpiryTeller__factory, IERC20__factory } from "src/typechain";
+import { ERC20BondToken__factory } from "src/typechain/factories";
+import { useAccount, useNetwork, useProvider } from "wagmi";
+
+/**
+ * Queries alchemy API for ERC_20 Token created topic and returns all events.
+ * Filters events by underlying asset == OHM
+ */
+export const useBondTokens = () => {
+  const { chain = { name: "mainnet", id: 1 } } = useNetwork();
+  const network = chain.name === "ethereum" ? "mainnet" : chain.name;
+  const networks = useTestableNetworks();
+  return useQuery<string[]>(["useBondToken"], async () => {
+    const options = {
+      method: "POST",
+      url: `https://eth-${network}.alchemyapi.io/v2/${process.env.REACT_APP_ETHEREUM_ALCHEMY_IDS}`,
```

Comment:
I think we can use Provider.getProviderURL here... let me double check

Reply:
fixed here: https://github.com/OlympusDAO/olympus-frontend/pull/2303

Reply:
i improved this further to just use the provider.getLogs directly.